apps/api_web/lib/api_web/controllers/route_pattern_controller.ex

```diff
@@ -93,24 +98,90 @@ defmodule ApiWeb.RoutePatternController do
 
   @spec format_filters(%{optional(String.t()) => String.t()}) :: RoutePattern.filters()
   defp format_filters(filters) do
-    Map.new(filters, fn {key, value} ->
+    filters
```

Comment:
suggestion: since some of the filters depend on each other, you might want to try a pattern like this:
```elixir
defp format_filters(filters, acc \\ %{})
defp format_filters(%{"route" => route_ids, "date" => date} = filters, acc) do
  # do the processing, add the filters to the acc
  new_filters = Map.drop(filters, ["route", "date"])
  format_filters(new_filters, new_acc)
end
defp format_filters(%{"id" => ids} = filters, acc) do
  new_acc = Map.put(acc, :ids, Params.split_on_comma(ids))
  new_filters = Map.delete(filters, "id")
  format_filters(new_filters, new_acc)
end
defp format_filters(_, acc) do
  acc
end
```